src/helpers/subPrograms.js

```diff
@@ -1,10 +1,22 @@
 const Program = require('../models/Program');
 const SubProgram = require('../models/SubProgram');
+const Step = require('../models/Step');
+const Activity = require('../models/Activity');
 
 exports.addSubProgram = async (programId, payload) => {
   const subProgram = await SubProgram.create(payload);
   await Program.updateOne({ _id: programId }, { $push: { subPrograms: subProgram._id } });
 };
 
-exports.updateSubProgram = async (subProgramId, payload) =>
-  SubProgram.updateOne({ _id: subProgramId }, { $set: payload });
+exports.updateSubProgram = async (subProgramId, payload) => {
+  if (!payload.status) return SubProgram.updateOne({ _id: subProgramId }, { $set: payload });
+
+  const subProgram = await SubProgram
+    .findOneAndUpdate({ _id: subProgramId }, { $set: payload })
+    .populate({ path: 'steps', select: 'activities' })
+    .lean();
+
+  await Step.updateMany({ _id: subProgram.steps.map(step => step._id) }, { status: payload.status });
```

Comment:
tu n'as pas beosin de mettre un `$in` ?

Reply:
J'ai l'impression qu'il n'y en avait pas besoin car ca fonctionnait sans mais je n'ai pas trouve de doc affirmant que ca fonctionnait bien donc je les ai ajouté :)